src/arithmetic/montgomery.rs

```diff
@@ -86,3 +86,159 @@ impl ProductEncoding for (RR, Unencoded) {
 impl ProductEncoding for (RR, RInverse) {
     type Output = <(RInverse, RR) as ProductEncoding>::Output;
 }
+
+#[allow(unused_imports)]
+use {
+    super::n0::N0,
+    crate::{bssl, c, limb::Limb},
+};
+
+#[cfg(not(any(
+    target_arch = "aarch64",
+    target_arch = "arm",
+    target_arch = "x86",
+    target_arch = "x86_64"
+)))]
+prefixed_export! {
+    unsafe fn bn_mul_mont(
+        r: *mut Limb,
+        a: *const Limb,
+        b: *const Limb,
+        n: *const Limb,
+        n0: &N0,
+        num_limbs: c::size_t,
+    ) {
+        use super::BIGINT_MODULUS_MAX_LIMBS;
+
+        // The mutable pointer `r` may alias `a` and/or `b`, so the lifetimes of
+        // any slices for `a` or `b` must not overlap with the lifetime of any
+        // mutable for `r`.
+
+        // Nothing aliases `n`
+        let n = unsafe { core::slice::from_raw_parts(n, num_limbs) };
+
+        let mut tmp = [0; 2 * MODULUS_MAX_LIMBS];
```

Comment:
I think this should be `BIGINT_MODULUS_MAX_LIMBS`, with this tweak it successfully builds and passes tests with `--no-default-features`.